src/binary_annotator.cpp

```diff
@@ -94,12 +102,50 @@ static bool IsNonZeroRegion(uint64_t offset, uint64_t length,
   return false;
 }
 
+static BinarySection GenerateMissingSection(const uint64_t offset,
```

Comment:
We don't always align to 8, so we should check smaller alignments too.

Also, we should confirm that `offset + length` is aligned.

Reply:
The check for < 8 is just to change the error message. I think we could probably update the logic here to consider more cases, but I am just covering the most extreme case that covers the other cases as well.

Reply:
Fair enough, consider leaving a comment or TODO then